### PR TITLE
Diagnostics sheet changes

### PR DESCRIPTION
**Pull request by jkadamczyk**
## What changed (plus any additional context for devs)

* Refactored the sheet to TS and to new file structure standard
* Added a state dump feature
* Added ability to copy UUID

### CODE REVIEW

**Review by skylarbarrera — APPROVED** [review]
this is fueg

smol UI ting on android but otherwise :shipit: 

https://cloud.skylarbarrera.com/Screenshot_20230421-102526.png

**Comment by jkadamczyk** [comment]
> this is fueg
> 
> smol UI ting on android but otherwise :shipit:
> 
> [cloud.skylarbarrera.com/Screenshot_20230421-102526.png](https://cloud.skylarbarrera.com/Screenshot_20230421-102526.png)

You mean the button right? I will try to get it fixed today. Thanks for letting me know

**Comment by jkadamczyk** [comment]
@SocketSecurity ignore rn-nodeify@10.2.0

File: src/screens/Diagnostics/helpers/createAndShareStateDumpFile.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import RNShare from 'react-native-share';
+import { IS_ANDROID } from '@/env';
+import { logger, RainbowError } from '@/logger';
+
+export async function createAndShareStateDumpFile() {
+  const appState = store.getState();
+  const stringifiedState = JSON.stringify(
+    appState,
+    getCircularReferenceReplacer()
+  );
```
Comment (estrattonbailey) [inline src/screens/Diagnostics/helpers/createAndShareStateDumpFile.ts:14]: Could just use flatted https://www.npmjs.com/package/flatted

**Review by estrattonbailey — APPROVED** [review]
Fueg is right 🔥

File: src/screens/Diagnostics/helpers/createAndShareStateDumpFile.ts (hunk 1 above)
Comment (jkadamczyk) [reply src/screens/Diagnostics/helpers/createAndShareStateDumpFile.ts:14]: I will merge it as is and open another that makes use of this lib, didn't know it existed

**Pull request merged**